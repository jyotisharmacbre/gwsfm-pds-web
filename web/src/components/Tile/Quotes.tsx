import React, { Component } from 'react';

class Quotes extends Component {
  render() {
    return (
      <div className="row">
        <div className="col-lg-12">
          <div className="card_outer_wrap quote_wrap">
            <div className="row pt-lg-4">
              <div className="col-lg-4 pl-md-2 pr-md-2">
                <div className="card_wrap">
                  <div className="card">
                    <h6 className="title">Quote 01</h6>
                    <form>
                      <div className="form-group">
                        <label>Supplier's Name</label>
                        <input type="text" placeholder="Enter Name" />
                      </div>
                      <div className="form-group">
                        <label>Quote Value (&#163;)</label>
                        <input type="text" placeholder="&#163;" />
                      </div>
                    </form>
                  </div>
                </div>
              </div>
              <div className="col-lg-4 pl-md-2 pr-md-2">
                <div className="card_wrap">
                  <div className="card">
                    <h6 className="title">Quote 02</h6>
                    <form>
                      <div className="form-group">
                        <label>Supplier's Name</label>
                        <input type="text" placeholder="Enter Name" />
                      </div>
                      <div className="form-group">
                        <label>Quote Value (&#163;)</label>
                        <input type="text" placeholder="&#163;" />
                      </div>
                    </form>
                  </div>
                </div>
              </div>
              <div className="col-lg-4 pl-md-2 pr-md-2">
                <div className="card_wrap">
                  <div className="card">
                    <h6 className="title">Quote 03</h6>
                    <form>
                      <div className="form-group">
                        <label>Supplier's Name</label>
                        <input type="text" placeholder="Enter Name" />
                      </div>
                      <div className="form-group">
                        <label>Quote Value (&#163;)</label>
                        <input type="text" placeholder="&#163;" />
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Quotes;
