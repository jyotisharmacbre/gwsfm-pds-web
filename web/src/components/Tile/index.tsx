import React from 'react';
import injectIntl from 'react-intl/dist/components/injectIntl';
import { Row, Col } from 'react-bootstrap';


export function Tile(props: any) {
  return (
    <Row>
      <Col lg={12}>
        <div className="card_outer_wrap">
          <Row className="pt-lg-3">
            <Col lg={4} className="pl-md-2 pr-md-2">
              <div className="card_wrap">
                
                <div className="card">
                  <h4 className="title">Preliminaries</h4>
                  <p className="text">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
                    do eiusmod tempor incididunt ut labore et dolore
                  </p>
                </div>
              </div>
            </Col>
            <Col lg={4} className="pl-md-2 pr-md-2">
              <div className="card_wrap">
                <div className="card">
                  <h4 className="title">Subcontractors</h4>
                  <p className="text">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
                    do eiusmod tempor incididunt ut labore et dolore
                  </p>
                </div>
              </div>
            </Col>
            <Col lg={4} className="pl-md-2 pr-md-2">
              <div className="card_wrap">
                <div className="card">
                  <h4 className="title">Discounts</h4>
                  <p className="text">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
                    do eiusmod tempor incididunt ut labore et dolore
                  </p>
                </div>
              </div>
            </Col>
          </Row>
        </div>
      </Col>
    </Row>
  );
}

export default injectIntl(Tile);
