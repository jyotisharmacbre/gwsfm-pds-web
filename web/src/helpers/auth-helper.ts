// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

