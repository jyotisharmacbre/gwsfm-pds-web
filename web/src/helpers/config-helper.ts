const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;