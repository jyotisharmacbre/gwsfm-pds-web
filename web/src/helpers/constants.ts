export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
