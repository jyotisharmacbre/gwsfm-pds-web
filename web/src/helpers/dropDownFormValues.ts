export const projectStatusData: object[] = [
  { label: '1', value: '1' },
  { label: '2', value: '2' },
  { label: '3', value: '3' }
];

export const engagementData: object[] = [
  { label: 'Type 1', value: '1' },
  { label: 'Type 2', value: '2' },
  { label: 'Type 3', value: '3' }
];

export const typesOfEngagement: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Fixed Price',
    value: 'Fixed Price'
  },
  {
    label: 'Time and Materials',
    value: 'Time and Materials'
  },
  {
    label: 'Risk and Reward',
    value: 'Risk and Reward'
  },
  {
    label: 'Cost Plus',
    value: 'Cost Plus'
  }
];

export const contractType: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'CT 1',
    value: 'CT 1'
  },
  {
    label: 'CT 2',
    value: 'CT 2'
  }
];

export const currency: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AED (UAE Dirham)',
    value: 'AED'
  },
  {
    label: 'AFN (Afghani)',
    value: 'AFN'
  },
  {
    label: 'ALL (Lek)',
    value: 'ALL'
  },
  {
    label: 'AMD (Dram)',
    value: 'AMD'
  },
  {
    label: 'ANG (Dutch Antilles Guilder)',
    value: 'ANG'
  },
  {
    label: 'AOA (Kwanza)',
    value: 'AOA'
  },
  {
    label: 'ARS (Peso)',
    value: 'ARS'
  }
];

export const projectStatus: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Initial Customer Inquiry',
    value: 'Initial Customer Inquiry'
  },
  {
    label: 'J&A',
    value: 'J&A'
  },
  {
    label: 'Archived',
    value: 'Archived'
  },
  {
    label: 'Lost',
    value: 'Lost'
  },
  {
    label: 'Order Received',
    value: 'Order Received'
  },
  {
    label: 'CPP',
    value: 'CPP'
  },
  {
    label: 'Complete',
    value: 'Complete'
  }
];

export const assetsWorkedOn: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AS 1',
    value: 'AS 1'
  },
  {
    label: 'AS 2',
    value: 'AS 2'
  }
];

export const enquiryTypeData: object[] = [
  { label: 'Written', value: '1' },
  { label: 'Verbal', value: '2' },
  { label: 'Full tender enquiry with specifications & orgs', value: '3' },
  { label: 'Other', value: '4' }
];
