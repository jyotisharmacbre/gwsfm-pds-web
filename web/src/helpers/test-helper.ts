import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
