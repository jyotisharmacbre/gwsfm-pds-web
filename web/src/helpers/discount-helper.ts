import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;