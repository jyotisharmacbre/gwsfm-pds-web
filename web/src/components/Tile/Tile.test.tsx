import React from 'react';
import { shallow } from 'enzyme';
import Tile from './index';
import {findByTestAtrr,checkProps} from '../../helpers/test-helper';

describe('card tile component', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(
        <Tile heading='heading-test' content='content-test'/>
    );
  });
  
  it('defines the component', () => {
    expect(wrapper).toBeDefined();
  }); 

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should Not throw a warning for proptypes', () => {
      const expectedProps = {
        heading: '',
       content: ''
      };
      const propsError = checkProps(Tile, expectedProps);
      expect(propsError).toBeUndefined();
    });

  it('should renders the heading', () => {
    expect(findByTestAtrr(wrapper,'card-heading')).toBeDefined();
  });

  it('should renders the content', () => {
    expect(findByTestAtrr(wrapper,'card-content')).toBeDefined();
  });
  
});
