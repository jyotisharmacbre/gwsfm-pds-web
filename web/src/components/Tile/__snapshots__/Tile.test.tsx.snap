// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card tile component should match the snapshot 1`] = `ShallowWrapper {}`;
